spec/features/create_goobi_image_spec.rb

```diff
@@ -0,0 +1,115 @@
+# frozen_string_literal: true
+
+require 'druid-tools'
+
+# NOTE: this spec will be skipped unless run on staging, since there is no goobi-qa
+RSpec.describe 'Create and accession object via Goobi', type: :feature,
+                                                        if: ENV.fetch('SDR_ENV', 'staging') == 'staging' do
+  let(:start_url) { "#{Settings.argo_url}/registration" }
+  let(:goobi_project_name) { 'Integration Testing' } # this project must exist in Goobi stage
+  let(:source_id_random_word) { "#{random_noun}-#{random_alpha}" }
+  let(:source_id) { "goobi-integration-test:#{source_id_random_word}" }
+  let(:label_random_words) { random_phrase }
+  let(:object_label) { "goobi integration test #{label_random_words}" }
+  let(:collection_name) { 'integration-testing' }
+  # this APO must exist on argo-stage and have goobiWF workflow and integration-testing collection available
+  let(:apo_name) { 'Goobi Testing APO' }
+  # the "Accession_Content_Expedited" workflow must exist on goobi-stage
+  let(:goobi_tag) { 'DPG : Workflow : Accession_Content_Expedited ' }
+  let(:goobi_workflow) { 'goobiWF' }
+
+  before do
+    authenticate!(start_url: start_url,
+                  expected_text: 'Register DOR Items')
+  end
+
+  after do
+    clear_downloads
+  end
+
+  scenario do
+    # register new object that will be sent to goobi
+    select apo_name, from: 'Admin Policy'
+    select collection_name, from: 'Collection'
+    select 'image', from: 'Content Type'
+    select goobi_workflow, from: 'Initial Workflow'
+    fill_in 'Project Name', with: goobi_project_name
+    fill_in 'Tags', with: goobi_tag
+
+    fill_in 'Source ID', with: source_id
+    fill_in 'Label', with: object_label
+
+    click_button 'Register'
+
+    # wait for object to be registered
+    expect(page).to have_text 'Items successfully registered.'
+
+    bare_object_druid = find('table a').text
+    druid = "druid:#{bare_object_druid}"
+
+    # puts druid # useful for debugging
+
+    # wait to be sure goobiWF has finished running and goobi has time to process the incoming object
+    sleep 2
```

Comment:
I'm curious which line below fails if we don't sleep here, and wondering if we can use a reload loop instead. It might also be a six-of-one-vs-half-dozen-of-other situation.

Reply:
basically if goobiWF has not finished running yet, then the goobi system will not yet aware of the object, and it won't be found (because the main task of goobiWF is to make a POST call out to the goobi system to register the object there)

instead of waiting, we could visit the detail page in argo and ensure the workflow is completed before proceeding, but unless it sometimes takes longer and this becomes flaky, that feels like more extra work (and likely just if not more actual time passed) than just pausing for a bit